torch/__init__.py

```diff
@@ -334,6 +334,11 @@ def compiled_with_cxx11_abi():
     return _C._GLIBCXX_USE_CXX11_ABI
 
 
+def compiled_with_BUILD_NAMEDTENSOR():
```

Comment:
does this have to be a public API (i.e., no underscores)?

Reply:
Nope, I'll make it private, thanks for pointing that out

Reply:
Should we also let user have a way to know it?

Reply:
I don't have any good ideas for how to make sure users don't use it aside from adding a docstring. I could just use `torch._C._BUILD_NAMEDTENSOR` directly and hope the double underscore is deterring enough...

Reply:
Oops, what I meant is that shouldn't we let user have a clean way to know whether the build can has named tensor built in...

Reply:
> Oops, what I meant is that shouldn't we let user have a clean way to know whether the build can has named tensor built in...

Sorry, I still don't understand the suggestion. The BUILD_NAMEDTENSOR flag is temporary and will be removed sometime in the future. A user should never have to check if they're using a "named tensor build" or a regular pytorch build.

Reply:
OK, I didn't know that. I vaguely remember that named tensor hurts performance a lot and that's why the build flag is there. If it's only temporary I don't see a problem.

Reply:
> named tensor hurts performance a lot and that's why the build flag is there.

The build flag is there to prevent any potential performance regressions while I throw the project together.